src/compiler/tsc.ts

```diff
@@ -415,7 +415,8 @@ namespace ts {
 
             const compileResult = compile(rootFileNames, compilerOptions, compilerHost);
 
-            if (!compilerOptions.watch) {
+            // Firefox has Object.prototype.watch
+            if (!compilerOptions.watch || !compilerOptions.hasOwnProperty("watch")) {

```

Comment:
renaming compiler options is technically a breaking change so I'd like to keep it as a last thing.